### ISSUE TITLE
Refactor Add labels

### ISSUE THREAD

**Issue by sagar87** [sagar87/spatialproteomics#40]
`pp.add_labels` should take `Mapping[cell -> label]` and add labels. The current implementation of `add_labels` should be renamed to `pp.add_labels_from_dataframe`.

**Comment by MeyerBender** [comment]
Right now, merging works as follows.

```
img = img.tl.cellpose()  # creates an n-dimensional segmentation mask in slow "_cellpose_segmentation"
img = img.pp.merge_segmentation("_cellpose_segmentation")  # creates a 1D segmentation mask in "_merged_segmentation" that has ct assignments in the attrs field
```

The idea would then be to have something like `img.pp.add_segmentation("_merged_segmentation", keep_labels=True)`, which would simultaneously add a segmentation and the corresponding labels.

**Comment by MeyerBender** [comment]
`add_segmentation()` currently has an argument called `relabel`, which will be confusing if we also have one called `keep_labels`. Will rename this to something like `reindex` (open for better suggestions).

**Comment by MeyerBender** [comment]
Addressed in #45, closing issue.
